app/src/main/java/org/simple/clinic/bloodsugar/entry/BloodSugarEntryEffectHandler.kt

```diff
@@ -68,9 +71,18 @@ class BloodSugarEntryEffectHandler @AssistedInject constructor(
         .addConsumer(SetBloodSugarReading::class.java, { ui.setBloodSugarReading(it.bloodSugarReading) }, schedulersProvider.ui())
         .addTransformer(UpdateBloodSugarEntry::class.java, updateBloodSugarEntryTransformer(schedulersProvider.io()))
         .addConsumer(ShowConfirmRemoveBloodSugarDialog::class.java, { ui.showConfirmRemoveBloodSugarDialog(it.bloodSugarMeasurementUuid) }, schedulersProvider.ui())
+        .addTransformer(LoadBloodSugarUnitPreference::class.java, loadBloodSugarUnitPreference())
         .build()
   }
 
+  private fun loadBloodSugarUnitPreference(): ObservableTransformer<LoadBloodSugarUnitPreference, BloodSugarEntryEvent> {
+    return ObservableTransformer { effect ->
+      effect
+          .switchMap { bloodSugarUnitPreference.asObservable().subscribeOn(schedulersProvider.io()) }
```

Comment:
I think we can observe on io scheduler here, instead of subscribing on io for the preference only.

```kt
      effect
          .observeOn(schedulersProvider.io())
          .switchMap { bloodSugarUnitPreference.asObservable() }
          .map(::BloodSugarUnitPreferenceLoaded)
```

Reply:
So we observe on the effect instead of observing just the preference, right?